File: files/2026-05-03/CSEC 03052026.py
import pandas as pd

path = "2026-05-03/Challenge 115.xlsx"
input = pd.read_excel(path, usecols="B:D", nrows=7, skiprows=2)
test = pd.read_excel(path, usecols="F:G", nrows=2, skiprows=2).rename(
	columns=lambda c: c.replace(".1", "")
)

result = (
	input.sort_values(["Product", "Date"])
	.assign(diff=lambda d: d.groupby("Product")["Price"].diff())
	.query("diff < 0")
	.drop_duplicates("Product")
	[["Product", "Date"]]
	.rename(columns={"Date": "First Drop Date"})
    .reset_index(drop=True)
)

print(result.equals(test))
# True

File: files/2026-05-03/CSEC 03052026.R
library(tidyverse)
library(readxl)

path <- "2026-05-03/Challenge 115.xlsx"
input <- read_excel(path, range = "B3:D9")
test <- read_excel(path, range = "F3:G5")

result = input %>%
  arrange(Product, Date) %>%
  filter(Price < lag(Price), .by = Product) %>%
  slice_head(n = 1, by = Product) %>%
  transmute(Product, `First Drop Date` = Date)

all.equal(result, test)
# [1] TRUE
